acceptance/tests/loader/autoload_from_resource_type_decl.rb

```diff
@@ -0,0 +1,84 @@
+        p "found_type_tst2"
+      end
+    end
+    END
+    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_type_1_path}", custom_type_1)
+    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_type_2_path}", custom_type_2)
+
+    site_pp = <<-PP
+    Resource['type_tst1'] { 'found_type': }
+    Resource['type_tst2'] { 'other_type_too': }
+    PP
+    create_sitepp(master, tmp_environment, site_pp)
+  end
+
+  on(master, "chmod -R 755 /tmp/#{tmp_environment}")
+
+  with_puppet_running_on(master, {}) do
+    agents.each do |agent|
+      on(agent, puppet("agent -t --server #{master.hostname} --environment #{tmp_environment}"),
+         :acceptable_exit_codes => 0) do |puppet_result|
+        assert_match(/Notice: \/File\[.*\/lib\/puppet\/type\/type_tst1\.rb\]\/ensure: defined content as /, puppet_result.stdout,
+                     "Expected to see notice for resource type: type_tst1")
+        assert_match(/Notice: \/File\[.*\/lib\/puppet\/type\/type_tst2\.rb\]\/ensure: defined content as /, puppet_result.stdout,
+                     "Expected to see notice for resource type: type_tst2")
+        assert_match(/found_type_tst1/, puppet_result.stdout, "Expected to see output from new type: type_tst1")
+        assert_match(/found_type_tst2/, puppet_result.stdout, "Expected to see output from new type: type_tst2")
+      end
+    end
+  end
+
+  step 'generate pcore files' do
+    on(master, puppet("generate types --environment #{tmp_environment}")) do |puppet_result|
+      assert_match(/Notice: Generating '.*\/type_tst1\.pp' using 'pcore' format/, puppet_result.stdout,
+                   "Expected to see Generating message for type: type_tst1")
+      assert_match(/Notice: Generating '.*\/type_tst2\.pp' using 'pcore' format/, puppet_result.stdout,
+                   "Expected to see Generating message for type: type_tst2")
+    end
+  end
+
+  # restart so that we will load and use the generated types
+  on(master, "service #{master['puppetservice']} restart")
+
+  agents.each do |agent|
+    step 'rerun agents after generate, ensure proper runs' do
+      on(agent, puppet("agent -t --server #{master.hostname} --environment #{tmp_environment}"),
+         :acceptable_exit_codes => 0) do |puppet_result|
+        assert_match(/found_type_tst1/, puppet_result.stdout, "Expected to see output from new type: type_tst1")
+        assert_match(/found_type_tst2/, puppet_result.stdout, "Expected to see output from new type: type_tst2")
+        refute_match(/Notice: \/File\[.*\/lib\/puppet\/type\/type_tst1\.rb\]\/ensure: defined content as /, puppet_result.stdout,
+                     "Unexpected notice for pre-exsiting generated type: type_tst1")
+        refute_match(/Notice: \/File\[.*\/lib\/puppet\/type\/type_tst2\.rb\]\/ensure: defined content as /, puppet_result.stdout,
+                     "Unexpected notice for pre-exsiting generated type: type_tst2")
+      end
+    end
```

Comment:
I'm missing tests that verifies that the master log does not contain "found_type_tst1". Testing that is important since it will show if the master loaded the ruby type or not. It's essential that the ruby type is not loaded on the master.

Please note that the agent is completely unaware of if the master loaded the ruby type or not. In order to check that, you must check the master log. It cannot be done as part of an `on(agent, ...)` step.

Perhaps you could do this.
1. Add an `on master` step, before the master is restarted, that asserts that the "found_type_tst1" is present in the master log.
2. Ensure that the master log is cleared when the master is restarted.
3. Add an `on master` step that executes last, and asserts that the master log does not contain "found_type_tst1".